src/server/sched_func.c

```diff
@@ -0,0 +1,288 @@
+ *
+ * PBS Pro is free software. You can redistribute it and/or modify it under the
+ * terms of the GNU Affero General Public License as published by the Free
+ * Software Foundation, either version 3 of the License, or (at your option) any
+ * later version.
+ *
+ * PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ * PARTICULAR PURPOSE.  See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Commercial License Information:
+ *
+ * The PBS Pro software is licensed under the terms of the GNU Affero General
+ * Public License agreement ("AGPL"), except where a separate commercial license
+ * agreement for PBS Pro version 14 or later has been executed in writing with Altair.
+ *
+ * Altair’s dual-license business model allows companies, individuals, and
+ * organizations to create proprietary derivative works of PBS Pro and distribute
+ * them - whether embedded or bundled with other software - under a commercial
+ * license agreement.
+ *
+ * Use of Altair’s trademarks, including but not limited to "PBS™",
+ * "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+ * trademark licensing policies.
+ *
+ */
+
+/**
+ * @file    sched_func.c
+ *
+ *@brief
+ * 		sched_func.c - various functions dealing with schedulers
+ *
+ */
+#include <errno.h>
+#include <string.h>
+#include "scheduler.h"
+#include "log.h"
+#include "pbs_ifl.h"
+#include "pbs_db.h"
+#include "pbs_error.h"
+#include <memory.h>
+#ifndef PBS_MOM
+extern pbs_db_conn_t	*svr_db_conn;
+#endif
+
+
+/**
+ * @brief
+ * 		sched_alloc - allocate space for a pbs_sched structure and initialize
+ *		attributes to "unset"
+ *
+ * @param[in]	sched_name	- scheduler  name
+ *
+ * @return	pbs_sched *
+ * @retval	null	- space not available.
+ */
+pbs_sched *
+sched_alloc(char *sched_name) {
+	int i;
+	pbs_sched *psched;
+
+	psched = (pbs_sched *) malloc(sizeof(pbs_sched));
+
+	if (psched == (pbs_sched *) 0) {
+		log_err(errno, "sched_alloc", "no memory");
+		return NULL;
+	}
+	(void) memset((char *) psched, (int) 0, (size_t) sizeof(pbs_sched));
+	CLEAR_LINK(psched->sc_link);
+	strncpy(psched->sc_name, sched_name, PBS_MAXSCHEDNAME);
+	psched->sc_name[PBS_MAXSCHEDNAME] = '\0';
+
+	append_link(&svr_allscheds, &psched->sc_link, psched);
+
+	/* set the working attributes to "unspecified" */
+
+	for (i = 0; i < (int) SCHED_ATR_LAST; i++) {
+		clear_attr(&psched->sch_attr[i], &sched_attr_def[i]);
+	}
+
+	return (psched);
+}
+
+/**
+ * @brief find a scheduler
+ *
+ * @param[in]	sched_name - scheduler name
+ *
+ * @return	pbs_sched *
+ */
+
+pbs_sched * find_scheduler(char *sched_name) {
+	char *pc;
+	pbs_sched *psched;
+
+	if (!strcmp(sched_name, "1") || !strcmp(sched_name,"sched"))
```

Comment:
I'm confused, in the database we add the default scheduler name as "main" and here it is referenced as "sched". only one of them can be true right?
Does this also needs to go to design document?